### PR TITLE
`<memory>`: Add different control block types for `allocate_shared_for_overwrite`

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
`allocate_shared_for_overwrite` creates allocated non-array objects by default-initialization, so these objects should be destroyed by plain destructor calls, not `Ator::destroy`. Currently the standard wording is unclear on `make_shared_for_overwrite` and `allocate_shared_for_overwrite`, so I submitted LWG-4024.

It's a bit unfortunate that the existing control block types can't be reused because the `_Destroy` functions can't recognize `_For_overwrite_tag`.

Unblocks one libcxx test:
- `std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.create/allocate_shared_for_overwrite.pass.cpp`

Drive-by changes:
- completing visualizers of control blocks for array types.

### CODE REVIEW

**Comment by StephanTLavavej** [comment]
Do these control blocks need new visualizers?

**Comment by frederick-vs-ja** [comment]
> Do these control blocks need new visualizers?

I'm now trying to add visualizer for `_Ref_count_obj_alloc_for_overwrite`.

But visualization of control block types for array types still seems missing and I have no idea how to add handle them. Previous PRs #309, #1315, and the previous issue #2787 didn't mention these stuffs.

File: stl/inc/memory [hunk 1]
```diff
@@ -2745,6 +2837,50 @@ private:
...
+
+template <class _Ty, class _Alloc>
+class _Ref_count_bounded_array_alloc_for_overwrite
+    : public _Ebco_base<_Rebind_alloc_t<_Alloc, remove_all_extents_t<_Ty>>>,
+      public _Ref_count_base {
+    // handle reference counting for bounded array in control block, allocator
+    // initialize and destroy objects by the default mechanism
+private:
+    static_assert(is_bounded_array_v<_Ty>);
+    static_assert(is_same_v<_Ty, remove_cv_t<_Ty>>, "allocate_shared_overwrite should remove_cv_t");
```
Comment (StephanTLavavej) [inline stl/inc/memory:2849]: Typo, should be `allocate_shared_for_overwrite`.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks! :heart_eyes_cat: This investigation, fix, and visualizers must have been a lot of work, I really appreciate it. I pushed a couple of tiny commits and I think this is ready to go.

**Pull request merged**